pkg/gadgets/run/tracer/run.go

```diff
@@ -792,28 +792,20 @@ func calculateColumnsForClient(
 		return nil, fmt.Errorf("getting value struct: %w", err)
 	}
 
-	colNames := map[string]struct{}{}
-
 	eventStruct, ok := gadgetMetadata.Structs[eventType.Name]
 	if !ok {
 		return nil, fmt.Errorf("struct %s not found in gadget metadata", eventType.Name)
 	}
 
-	for _, field := range eventStruct.Fields {
-		colNames[field.Name] = struct{}{}
+	members := map[string]btf.Member{}
+	for _, member := range eventType.Members {
+		members[member.Name] = member
 	}
 
 	columns := []types.ColumnDesc{}
 
-	for _, member := range eventType.Members {
-		member := member
-
-		_, ok := colNames[member.Name]
-		if !ok {
-			logger.Debugf("field %q not present on metadata file, skipping", member.Name)
-			continue
-		}
-
+	for _, field := range eventStruct.Fields {
+		member := members[field.Name]
```

Comment:
Can it happen that the field from the metadata doesn't exist as member of the btf type?

Reply:
Good point. Yes, it could happen if both the gadget is built and run with `--validate-metadata=false`. I added an additional check for it.